app/controllers/thread_controller.php

```diff
@@ -80,4 +113,124 @@ public function create()
         $this->render($page);
     }
+
+    public function edit()
+    {
+        if (!isset($_SESSION['username'])) {
+            redirect(url('/'));
+        }
+        $check = Param::get('check', false);
+        $title = " | Edit thread";
+        $user = new User;
+        $user->setInfoByUsername($_SESSION['username']);
+        $params = array(
+            'thread_id' => Param::get('id', 0),
+            'user_id'   => $user->user_id,
+        );
+        $thread = new Thread($params);
+
+        $thread_content = Thread::get(Param::get('id', 0));
+        $thread->error = isset($thread_content->error) ? $thread_content->error : null;
+        if (isset($thread->error)) {
+            $this->set(get_defined_vars());
+            return;
+        }
+        if ($thread_content->user_id !== $user->user_id) {
+            $thread->error = 'Cannot edit other user\'s thread';
+            $this->set(get_defined_vars());
+            return;
+        }
+        if ($check) {
+            $thread->title = Param::get('new_thread', '');
+            try {
+                $thread->editThread();
+            } catch (ValidationException $e) {
+                $thread->error = 'Input Error, Please enter at from 1 to 200 charcters';
+            } catch (Exception $e) {
+                $thread->error = 'Unexpected Error occured';
+            }
+        }
+        $this->set(get_defined_vars());
+    }
+
+    public function top_threads()
+    {
+        if (!isset($_SESSION['username'])) {
+            redirect(url('/'));
+        }
+        $type = Param::get('type', '');
+        if ($type === '') {
+            redirect(url('/'));
+        }
+        if ($type === 'comment') {
+            $limit = getLimit(Comment::getThreadsByCommentCount()); 
+        } else {
+            $limit = getLimit(Follow::getThreadsByFollowCount()); 
+        }
+        switch ($type) {
+            case 'comment':
+                $top_threads = Thread::getMostCommented($limit);
+                $title = 'Most Commented Thread';
+                $sub_title = "Showing top {$limit} most commented threads";
+                break;
+            case 'follow':
+                $top_threads = Thread::getMostFollowed($limit);
+                $title = 'Most Followed Thread';
+                $sub_title = "Showing top {$limit} most followed threads";
+                break;
+            default:
+                break;
+        }
+        $user = new User;
+        $user->setInfoByUsername($_SESSION['username']);
+        $this->set(get_defined_vars());
+    }
+
+    public function user_thread()
+    {
+        if (!isset($_SESSION['username'])) {
+            redirect(url('/'));
+        }
+        $user = new User($_SESSION['username']);
+        $user_id = Param::get('user_id', $user->user_id);
+        $url_params = '?user_id='.$user_id.'&';
+        $title = $user->username . '\'s Threads';
+        $page = Param::get('page', 1);
+        $per_page = 5;
```

Comment:
similar comment to : https://github.com/j-asis/Web-Portal/pull/2/files#r28211496
(apply to all others :bow: )